internal/backend/dispatcher/eventsworkflow.go

```diff
@@ -73,11 +73,20 @@ func (d *dispatcher) getEventSessionData(ctx context.Context, event sdktypes.Eve
 
 		z := z.With(zap.String("env_id", envID.String()), zap.String("event_type", triggerEventType))
 
-		if triggerEventType != eventType {
+		if triggerEventType != "" && eventType != triggerEventType {
 			z.Debug("irrelevant event type")
 			continue
 		}
 
+		if relevant, err := event.Matches(t.Filter()); err != nil {
+			z.Debug("filter error", zap.Error(err))
+			// TODO: alert user their filter is bad. Integrate with alerting and monitoring.
```

Comment:
Is this an actual task, or "someone needs to think about this sometime in the future"? If it's the first, please add an issue ID.

Either way, alerting and monitoring are for the server, not for the user, no? What if this is happening on AK cloud?

Reply:
these failures might be a result of a misconfiguration by the user, even in multi tenant when the filter is bad. we should alert the user (consumer in this case?) that they should check it. will add an issue.